cli/command/utils.go

```diff
@@ -117,3 +118,18 @@ func PruneFilters(dockerCli Cli, pruneFilters filters.Args) filters.Args {
 
 	return pruneFilters
 }
+
+// PlatformFromEnvironment sets the platform from an environment variable
+// if it isn't otherwise populated.
+func PlatformFromEnvironment(platform string) string {
```

Comment:
Since this is effectively setting a default value for platform I think the environment variable should be read as the default value in `AddPlatformFlag()` instead of exporting this function:

```go
flags.StringVar(target, "platform", os.Getenv("DOCKER_DEFAULT_PLATFORM"), ...)
```

Reply:
That makes sense. Updated.

For hiding the flag, it is going to require getting consensus on https://github.com/moby/moby/pull/34628 first, which I don't think there is among other maintainers yet.